### PR TITLE
feat: automatically generate, create and update library repos

### PR DESCRIPTION
**Pull request by Duologic**
This PR adds the last bits to automate all of this through Github Actions:

- Create repositories for all libraries with Terraform
- Per library:
  - Clone the repository
  - Remove all but `.git`
  - Generate the library and docs
  - Add, Commit and Push

This PR also does the rename `s/k8s-alpha/k8s-libsonnet` and takes the common approach of the `-libsonnet` suffix as discussed/voted for in the #tanka channel on Grafana's community Slack.

TODO before merge:
- [x] Figure out where to store the Terraform state

TODO after merge:
- [ ] Update `k8s-alpha` to use `k8s-libsonnet` and emit a deprecation notice through `std.trace()`
- [ ] Remove or update `cert-manager-lib` in the same way
- [ ] Rename this repository

Ideas for improvement:
- Improve the index READMEs with instructions to install through `jb`, links to Tanka, ...
- Set up "hub" that displays the current libraries, integrate with https://artifacthub.io/ ?
- Add mooorrrreee libraries

### CODE REVIEW

**Comment by Duologic** [comment]
I have created a Terraform Cloud organization to store the TF state, feel free to bug me if you need access.

**Comment by xvzf** [comment]
First glance looks really good! :+1:

File: Dockerfile [hunk 1]
```diff
@@ -1,8 +1,3 @@
-FROM bitnami/kubectl:1.21.1 as kubectl
-FROM rancher/k3s:v1.21.1-k3s1 as k3s
-FROM mikefarah/yq:4.9.3 as yq2
-FROM jsonnetlibs/docsonnet:0.0.1 as docsonnet
-
 FROM golang:1.16.4 as base
```
Comment (julienduchesne) [inline Dockerfile:1]: You can bump that to 1.16.5
```suggestion
FROM golang:1.16.5 as base
```

File: jsonnet/github_action.jsonnet [hunk 2]
```diff
@@ -1,25 +1,69 @@
...
+                chmod 600 ~/.ssh/id_rsa
+                export GEN_COMMIT=1
+              |||,
+            },
+            { run: 'make build libs/' + lib.name },
+          ],
+        }
+        for lib in libs
+      } + {
+        what_is_this: {
```
Comment (julienduchesne) [inline jsonnet/github_action.jsonnet:30]: Forgot this?

File: jsonnet/github_action.jsonnet [hunk 3]
```diff
@@ -1,25 +1,69 @@
...
+            { run: 'echo isPR? ' + onPR['if'] },
+            { run: 'echo ${{ github.repository }}' },
+            { run: 'echo ${{ github.ref }}' },
+            { run: 'echo ${{ github.event_name }}' },
+          ],
+        },
+        terraform: {
+          local tf_env = {
+            'working-directory': 'tf',
+            env: {
```
Comment (julienduchesne) [inline jsonnet/github_action.jsonnet:44]: To have a nicer output, I'd also set `TF_IN_AUTOMATION` to true: https://www.terraform.io/docs/cli/config/environment-variables.html#tf_in_automation

**Review by julienduchesne — APPROVED** [review]
Should this repository be renamed to `lib-gen` or something? It doesn't have much to do with k8s anymore

This all LGTM! Good job!

File: Dockerfile (hunk 1 above)
Comment (Duologic) [reply Dockerfile:1]: This is impossible to keep track of 😌

File: jsonnet/github_action.jsonnet (hunk 2 above)
Comment (Duologic) [reply jsonnet/github_action.jsonnet:30]: Nah, left it on purpose.

**Comment by Duologic** [comment]
> Should this repository be renamed to `lib-gen` or something?

Planning that in a follow up PR

File: Dockerfile (hunk 1 above)
Comment (Duologic) [reply Dockerfile]: I dropped the patch part.

File: jsonnet/github_action.jsonnet (hunk 2 above)
Comment (Duologic) [reply jsonnet/github_action.jsonnet]: Renamed it so it looks like it belongs there.

**Pull request merged**